src/machine/machine_nrf52xxx.go

```diff
@@ -164,13 +168,22 @@ func (a ADC) Get() uint16 {
 	}
 	nrf.SAADC.EVENTS_STOPPED.Set(0)
 
-	value := int16(rawValue.Get())
-	if value < 0 {
-		value = 0
+	rawValue := rawValue.Get()
+	// convert to 16 bit resolution/value
+	switch nrf.SAADC.RESOLUTION.Get() {
+	case nrf.SAADC_RESOLUTION_VAL_8bit:
+		rawValue <<= 8
+	case nrf.SAADC_RESOLUTION_VAL_10bit:
+		rawValue <<= 6
+	case nrf.SAADC_RESOLUTION_VAL_12bit:
+		rawValue <<= 4
+	case nrf.SAADC_RESOLUTION_VAL_14bit:
+		rawValue <<= 2
+	default:
+		rawValue <<= 4 // 12bit
```

Comment:
The switch case can be replaced by:
```golang
	// max precision 14bit (<<= 2, for 16 bit value)
	// min precision 8bit (<<= 8, for 16 bit value)
	// resolution constants 8bit ~ 0, 14bit ~ 3
	precisionAdjustment := 8 - 2*nrf.SAADC.RESOLUTION.Get()
        rawValue <<= precisionAdjustment
```
its more compact, but also "magical". Just tell me what you prefer.

Reply:
I like this simple linear function conversion instead of a multiline switch 👍